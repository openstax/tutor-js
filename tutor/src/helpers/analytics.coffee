_ = require 'underscore'

Routes = require '../routes'

{CurrentUserStore} = require '../flux/current-user'

DestinationHelper  = require '../helpers/routes-and-destinations'

# generate custom event data for routes
Events =
  viewTaskStep: ({courseId}) ->
    # track all work done on a course
    Analytics.sendEvent 'Course', 'Work', label: courseId

  viewStudentDashboard: ({courseId}) ->
    # compare activity between courses
    Analytics.sendEvent 'Student', 'Dashboard', label: courseId


# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
Translators =

  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
  viewPerformanceForecast:  ({courseId}) -> "/student/performance-forecast/#{courseId}"
  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"

  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
  viewTeacherPerformanceForecast:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
  editReading:           ({courseId}) -> "/teacher/assignment/edit/reading/#{courseId}"
  editHomework:          ({courseId}) -> "/teacher/assignment/edit/homework/#{courseId}"
  editExternal:          ({courseId}) -> "/teacher/assignment/edit/external/#{courseId}"
  createReading:         ({courseId}) -> "/teacher/assignment/create/reading/#{courseId}"
  createHomework:        ({courseId}) -> "/teacher/assignment/create/homework/#{courseId}"
  createExternal:        ({courseId}) -> "/teacher/assignment/create/external/#{courseId}"
  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"

  viewReferenceBook:        ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookSection: ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookPage:    ({courseId}) -> "/reference-view/#{courseId}"

  # Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep:         ({courseId}) ->
    role = CurrentUserStore.getCourseRole(courseId, true)
    "/#{role}/task-step/#{courseId}"


GA = undefined

Analytics =

  setTracker: (tracker) -> GA = tracker

  sendPageView: (url) ->
    GA?('send', 'pageview', url)

  onNavigation: (path) ->
    return unless GA

    route = Routes.pathToEntry(path)

    return @sendPageView("/not-found/#{change.path}") unless route

    path = Translators[route.entry.name]?( route.match.params ) or route.match.pathname

    # if we're also going to send custom events then we set the page
    if Events[route.entry.name]
      GA('set', 'page', path)
      Events[route.entry.name]( route.match.params )
      @sendPageView() # url's not needed since it was set before events
    else
      @sendPageView(route.match.pathname)

  sendEvent: (category, action, attrs) ->
    return unless GA
    GA('send', 'event', category, action, attrs.label, attrs.value)

module.exports = Analytics
