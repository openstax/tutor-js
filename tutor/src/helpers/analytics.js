import { partial, get } from 'lodash';

import Router from '../helpers/router';
import Courses from '../models/courses-map';

// generate custom event data for routes
const Events = {
  viewTaskStep({ courseId }) {
    // track all work done on a course
    return Analytics.sendEvent('Course', 'Work', { label: courseId });
  },

  viewStudentDashboard({ courseId }) {
    // compare activity between courses
    return Analytics.sendEvent('Student', 'Dashboard', { label: courseId });
  },
};

// a bit shorter helper methods
const isTeacher = courseId => get(Courses.get(courseId), 'currentRole.isTeacher', false);
const getRole = function(courseId) {
  if (Courses.get(courseId).is_teacher) { return 'teacher'; } else { return 'student'; }
};

const assignmentTypeTranslator = function(assignmentType, { courseId, id }) {
  const newOrEdit = id === 'new' ? 'create' : 'edit';
  return `/teacher/assignment/${newOrEdit}/${assignmentType}/${courseId}`;
};

function viewReferenceBook({ courseId, pageId }) {
  const url = `/reference-view/${courseId}`;
  if (!pageId) {
    return url;
  }
  const course = Courses.get(courseId);
  const page = course.referenceBook.pages.byId.get(pageId);
  return page ? `${url}/section/${page.chapter_section.key}` : url;
}

// Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
const Translators = {
  dashboard({ courseId }) {
    if (isTeacher(courseId)) { return `/teacher/calendar/${courseId}`; } else { return `/student/dashboard/${courseId}`; }
  },
  practiceTopics({ courseId }) { return `/student/practice/${courseId}`; },
  viewPerformanceGuide({ courseId }) {
    return `/${getRole(courseId)}/performance-forecast/${courseId}`;
  },
  calendarByDate({ courseId }) { return `/teacher/calendar/${courseId}`; },
  viewGradebook({ courseId }) { return `/teacher/student-scores/${courseId}`; },
  courseSettings({ courseId }) { return `/teacher/roster/${courseId}`; },
  editAssignment: partial(assignmentTypeTranslator, 'assignment'),
  calendarViewPlanStats({ courseId }) { return `/teacher/metrics/quick/${courseId}`; },
  reviewTask({ courseId }) { return `/teacher/metrics/review/${courseId}`; },
  viewReferenceBookSection: viewReferenceBook,
  viewReferenceBookPage: viewReferenceBook,
  viewReferenceBook,

  // Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep({ courseId }) {
    const course = Courses.get(courseId);
    const role = course ? course.primaryRole.type : 'unknown';
    return `/${role}/task-step/${courseId}`;
  },
};

const TAG_MANAGER_ID = 'GTM-W6N7PB';

let GA = undefined;
let trackerNames = undefined;

var Analytics = {
  // copied from BIT
  initTagManager(w, d, s, l, i) {
    // Disable ESLint rules since we're copying Google's script
    /* eslint-disable one-var, prefer-const, prefer-template */
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    let f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l !== 'dataLayer' ? '&l=' + l : '';
    // Breaks in tests because there are no scripts
    if (f) {
      j.async = true;
      j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl;
      f.parentNode.insertBefore(j, f);
    }
  },
  
  // Set the Command Queue (ga) and cache the (possibly multiple) tracker names
  setGa(ga) {
    trackerNames = undefined;
    GA = ga;
    if (GA) {
      GA(function() {
        trackerNames = GA.getAll().map(function(tracker) { return tracker.get('name'); });
      });
      this.initTagManager(window, document, 'script', 'dataLayer', TAG_MANAGER_ID);
    }

    return GA;
  },

  // Calls all trackers with the command and arguments given here
  ga(command, ...params) {
    if (!GA || !trackerNames) { return; }
    trackerNames.forEach((trackerName) => {
      this.realGa(trackerName + '.' + command, ...params);
    });
  },

  // Exists largely for spec mocking
  realGa(...args) {
    GA(...args);
  },

  sendPageView(url) {
    this.ga('send', 'pageview', url);
  },

  recordCourseDimension(courseId) {
    const course = Courses.get(courseId);
    if (course) {
      this.ga('set', 'dimension1', course.appearance_code);
    }
  },

  onNavigation(path) {
    if (!GA) { return; }
    const route = Router.currentMatch(path);

    if (!route) {
      this.sendPageView(`/not-found${path}`);
      return;
    }
    const { courseId } = route.params;
    if (courseId) { this.recordCourseDimension(courseId); }
    const translatedPath = Translators[route.entry.name] ?
      Translators[route.entry.name]( route.params ) : route.pathname;

    this.ga('set', 'page', translatedPath);
    // if we're also going to send custom events then we set the page
    if (Events[route.entry.name]) {
      Events[route.entry.name]( route.params );
    }

    this.sendPageView(); // url's not needed since it was set before events
  },

  sendEvent(category, action, attrs = {}) {
    this.ga('send', 'event', category, action, attrs.label, attrs.value);
  },
};

export default Analytics;
