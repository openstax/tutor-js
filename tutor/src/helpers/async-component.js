import React from 'react';
import AsyncErrorBoundary from '../components/error-monitoring/async-load-error';
import LoadingAnimation from '../../../shared/src/components/loading-animation'

export function asyncComponent(loader, name = '') {

    const Component = React.lazy(loader);
    const loading = <LoadingAnimation message={`Loading ${name}â€¦`} />;

    const Loader = (props) => (
        <AsyncErrorBoundary>
            <React.Suspense fallback={loading}>
                <Component {...props} />
            </React.Suspense>
        </AsyncErrorBoundary>
    );

    return (props) => <Loader {...props} />;
}


export function loadAsync(resolve, name) {
    // return a function so the router will only evaluate it when it's needed
    // in the future we may insert role dependant logic here
    return () => asyncComponent(resolve, name);
}
