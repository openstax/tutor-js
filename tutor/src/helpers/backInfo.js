import { navHistory } from '../models'


const backInfo = () => {
    // Gets route to last path that was not the same as the current one
    // See TransitionStore for more detail.
    const historyInfo = navHistory.latest

    const text = historyInfo != null && !!historyInfo.name ? `Back to ${historyInfo.name}` : 'My Courses';
    const to =  historyInfo != null && !!historyInfo.path ? historyInfo.path : 'myCourses';

    return {
        text,
        to,
    };
};


export default backInfo;
