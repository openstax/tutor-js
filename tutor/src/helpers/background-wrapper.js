import { styled } from 'vendor';
import { navbars } from '../theme.js';

const BackgroundWrapper = styled.div`
  background: #fff;
  min-height: calc(100vh - ${navbars.top.height} - ${navbars.bottom.height});
  position: relative;
  overflow: hidden;
  padding: 0 2.4rem;
`;

const ContentWrapper = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

export { BackgroundWrapper, ContentWrapper };
