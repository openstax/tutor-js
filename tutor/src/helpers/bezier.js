function EASE_IN_OUT(t) {
    if (t < .5) { return 4 * t * t * t; } else { return ((t - 1) * ((2 * t) - 2) * ((2 * t) - 2)) + 1; }
}

function CALC(start, end, elapsed, duration) {
    if (elapsed > duration) { return end; }
    return start + ((end - start) * EASE_IN_OUT(elapsed / duration));
}

export default function bezier({
    range, duration, onStep, onComplete,
}) {
    const startTime = Date.now();
    const requestAnimationFrame = window.requestAnimationFrame || window.setTimeout;

    const step = () => {
        const elapsed = Date.now() - startTime;
        onStep(CALC(range[0], range[1], elapsed, duration));
        if (elapsed < duration) {
            requestAnimationFrame(step);
        } else if (onComplete) {
            onComplete();
        }
    };

    step();
}
