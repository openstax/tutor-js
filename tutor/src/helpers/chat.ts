import { Raven } from '../models'

const DEPLOYMENT_ID   = '5724P000000ghju';
const ORGANIZATION_ID = '00DU0000000Kwch';
const ELEMENT_ID      = '5736f000000k9eQ';


const win = window as any

export const Chat = {
    initialized: false,

    get isEnabled() {
        return Boolean(win.liveagent);
    },

    get isOnline() {
        if (!this.isEnabled) { return false; }
        const chat = document.querySelector<HTMLDivElement>('.chat.enabled');
        return chat && chat.style.display !== 'none';
    },

    initialize() {
    // will emit debugging info to the console
    // liveagent.enableLogging();
    },

    setElementVisiblity(online: any, offline?: any) {
        if (!this.isEnabled) { return; }
        try {
            win.liveagent.showWhenOnline(ELEMENT_ID, online);
            if (offline) {
                win.liveagent.showWhenOffline(ELEMENT_ID, offline);
            }
            if (!this.initialized) {
                win.liveagent.init('https://d.la1-c2-ia5.salesforceliveagent.com/chat', DEPLOYMENT_ID, ORGANIZATION_ID);
                this.initialized = true;
            }
        } catch(err) {
            Raven.captureException(err);
        }
    },

    start() {
        win.liveagent.startChat(ELEMENT_ID);
    },

};
