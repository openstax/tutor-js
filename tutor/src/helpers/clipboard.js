import _ from 'underscore';

const SUPPORTED = typeof document.queryCommandSupported === 'function' ? document.queryCommandSupported('copy') : undefined;

export default {
    isSupported: _.constant(!!SUPPORTED),

    copy() {
        if (!SUPPORTED) { return; }

        try {
            document.execCommand('copy');
        } catch (e) {
            // eslint-disable-next-line no-console
            console.warn('clipboard copy failed', e);
        }
    },
};
