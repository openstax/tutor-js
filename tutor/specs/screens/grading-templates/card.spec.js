import { React, Factory } from '../../helpers';
import Card from '../../../src/screens/grading-templates/card';

describe('Grading Templates Card', function() {
    let props;

    beforeEach(() => {
        const template = Factory.gradingTemplate({ task_plan_type: 'reading' });

        props={
            template,
            onEdit: jest.fn(),
            onDelete: jest.fn(),
        };
    });

    it('matches snapshot for reading', () => {
        expect.snapshot(<Card {...props} />).toMatchSnapshot();
    });

    it('matches snapshot for homework', () => {
        props.template = Factory.gradingTemplate({ task_plan_type: 'homework' });
        expect.snapshot(<Card {...props} />).toMatchSnapshot();
    });

    it('deletes and edits', () => {
        const card = mount(<Card {...props} />);
        // cant delete unless there is a matching template
        expect(card.find('button.ox-icon-trash')).toHaveLength(0);
        props.template.map = {
            array: { find() { return props.template; } },
            course: Factory.course(),
        };
        expect(props.template.canRemove).toBe(true);
        card.update();
        card.find('button.ox-icon-trash').simulate('click');
        expect(props.onDelete).toHaveBeenCalledWith(props.template);

        card.find('button.ox-icon-edit').simulate('click');
        expect(props.onEdit).toHaveBeenCalledWith(props.template);
    });
});
