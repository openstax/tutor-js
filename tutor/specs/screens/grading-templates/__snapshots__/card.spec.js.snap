// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grading Templates Card matches snapshot for homework 1`] = `
.c2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #424242;
  line-height: 2rem;
  height: 100%;
}

.c1 .card {
  border: 1px solid #d5d5d5;
  height: 100%;
}

.c1 .card-header {
  border: 0;
  border-left: 8px solid #34bdd7;
  background: #daf3f8;
  font-size: 1.6rem;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .card-header .btn:not(.btn-link):focus {
  box-shadow: none;
}

.c1 .card-header .btn-link {
  color: #424242;
  padding: 0;
}

.c3 {
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: bold;
  margin-bottom: 0.8rem;
}

.c4 {
  color: #929292;
}

.c5 {
  color: #5f6163;
  font-weight: bold;
  margin-bottom: 0.8rem;
}

.c6 {
  border-bottom: 1px solid #d5d5d5;
  margin-bottom: 0.8rem;
}

.c0 {
  margin-top: 3rem;
}

<div
  className="c0 col-lg-4 col-md-6 col-sm-12 col-12"
>
  <div
    className="c1"
    data-id={3}
    data-test-id="grading-template-card"
    data-type="homework"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Default Homework
        <div>
          <button
            className="ox-icon-edit btn btn-plain"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-edit fa-w-18 ox-icon ox-icon-edit c2"
              data-icon="edit"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h4
          className="c3"
        >
          SHOW SCORES & FEEDBACK TO STUDENTS
        </h4>
        <div
          className="c4"
        >
          For auto-graded questions:
        </div>
        <div
          className="c5"
        >
          invalid value
        </div>
        <div
          className="c4"
        >
          For manually-graded questions:
        </div>
        <div
          className="c5"
        >
          invalid value
        </div>
        <div
          className="c6"
        />
        <h4
          className="c3"
        >
          LATE WORK POLICY
        </h4>
        <div
          className="c4"
        >
          Accept late work?
        </div>
        <div
          className="c5"
        >
          Yes
        </div>
        <div
          className="c4"
        >
          Late work penalty:
        </div>
        <div
          className="c5"
        >
          Deduct 
          10%
           for each late day
        </div>
        <div
          className="c6"
        />
        <h4
          className="c3"
        >
          DEFAULT DATE AND TIME
        </h4>
        <div
          className="c4"
        >
          Due date for assignments:
        </div>
        <div
          className="c5"
        >
          1
           days after open date
        </div>
        <div
          className="c4"
        >
          Due time for assignments:
        </div>
        <div
          className="c5"
        >
          5:00 PM
        </div>
        <div
          className="c4"
        >
          Close date for assignments:
        </div>
        <div
          className="c5"
        >
          1
           days after due date
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Grading Templates Card matches snapshot for reading 1`] = `
.c2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #424242;
  line-height: 2rem;
  height: 100%;
}

.c1 .card {
  border: 1px solid #d5d5d5;
  height: 100%;
}

.c1 .card-header {
  border: 0;
  border-left: 8px solid #f4d018;
  background: #fcf6d8;
  font-size: 1.6rem;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .card-header .btn:not(.btn-link):focus {
  box-shadow: none;
}

.c1 .card-header .btn-link {
  color: #424242;
  padding: 0;
}

.c3 {
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: bold;
  margin-bottom: 0.8rem;
}

.c4 {
  color: #929292;
}

.c5 {
  color: #5f6163;
  font-weight: bold;
  margin-bottom: 0.8rem;
}

.c6 {
  border-bottom: 1px solid #d5d5d5;
  margin-bottom: 0.8rem;
}

.c0 {
  margin-top: 3rem;
}

<div
  className="c0 col-lg-4 col-md-6 col-sm-12 col-12"
>
  <div
    className="c1"
    data-id={1}
    data-test-id="grading-template-card"
    data-type="reading"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Default Reading
        <div>
          <button
            className="ox-icon-edit btn btn-plain"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-edit fa-w-18 ox-icon ox-icon-edit c2"
              data-icon="edit"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h4
          className="c3"
        >
          SCORE CALCULATION FOR QUESTIONS
        </h4>
        <div
          className="c4"
        >
          Weight for correctness:
        </div>
        <div
          className="c5"
        >
          77%
           of question’s point value
        </div>
        <div
          className="c4"
        >
          Weight for completion:
        </div>
        <div
          className="c5"
        >
          64%
           of question’s point value
        </div>
        <div
          className="c6"
        />
        <h4
          className="c3"
        >
          LATE WORK POLICY
        </h4>
        <div
          className="c4"
        >
          Accept late work?
        </div>
        <div
          className="c5"
        >
          Yes
        </div>
        <div
          className="c4"
        >
          Late work penalty:
        </div>
        <div
          className="c5"
        >
          Deduct 
          10%
           for each late day
        </div>
        <div
          className="c6"
        />
        <h4
          className="c3"
        >
          DEFAULT DATE AND TIME
        </h4>
        <div
          className="c4"
        >
          Due date for assignments:
        </div>
        <div
          className="c5"
        >
          1
           days after open date
        </div>
        <div
          className="c4"
        >
          Due time for assignments:
        </div>
        <div
          className="c5"
        >
          5:00 PM
        </div>
        <div
          className="c4"
        >
          Close date for assignments:
        </div>
        <div
          className="c5"
        >
          1
           days after due date
        </div>
      </div>
    </div>
  </div>
</div>
`;
